pallets/portfolio/src/lib.rs

```diff
@@ -362,7 +362,7 @@ decl_module! {
             });
             storage_migrate_on!(StorageVersion, 2, {
                 Portfolios::iter()
-                    .filter(|(identity, number, name)| number == &Self::name_to_number(identity, name))
+                    .filter(|(identity, number, name)| Some(number) == Self::name_to_number(identity, name).as_ref())
                     .for_each(|(identity, number, name)| {
                             NameToNumber::insert(identity, name, number);
                     }
```

Comment:
We should start deleting old migration code like this.

Once the migrations have run on Mainnet, we don't need to keep them.  They just waste space in the runtime wasm binary.